### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by MNisarAli**
## 🏷️ Adding Species Column and Transactions to Animals Table

This pull request contains changes to the animals table, including the addition of a species column and various transaction operations. Here's a summary of what's new:

### Changes Made 🔄:

- Added a species column of type string to the animals table and modified the `schema.sql` file.
- Inserted seven animals with their details, including their name, birth date, weight, neutered status, and escape attempts.
- Updated the animals table inside a transaction to set the species column to unspecified and then rolled back the change to verify the original state was restored.
- Updated the animals table inside a transaction to set the species column to "digimon" for animals with names ending in "mon" and "pokemon" without a species set, and committed the transaction.
- Deleted all records in the animals table inside a transaction and then rolled back the change to verify the original state was restored.
- Deleted all animals born after Jan 1st, 2022, and created a savepoint for the transaction. Updated all animals' weights to be their weight multiplied by -1, rolled back to the savepoint, and then updated all animals' weights that are negative to be their weight multiplied by -1. Finally, committed to the transaction.
- Created queries per [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#project-requirements)

### Screenshots 📷:

image
image
image
image
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,4 +5,11 @@ VALUES
 ('1', 'Agumon', '2020-02-03', '0', 'true', '10.23'),
 ('2', 'Gabumon', '2018-11-15', '2', 'true', '8'),
 ('3', 'Pikachu', '2021-01-07', '1', 'false', '15.04'),
-('4', 'Devimon', '2017-05-12', '5', 'true', '11');
+('4', 'Devimon', '2017-05-12', '5', 'true', '11')
+('5', 'Charmander', '2020-02-08', '0', 'false', '-11'),
+('6', 'Plantmon', '2021-11-15', '2', 'true', '-5.7'),
+('7', 'Squirtle', '1993-04-02', '3', 'false', '-12.13'),
+('8', 'Angemon', '2005-06-12', '1', 'true', '-45'),
+('9', 'Boarmon', '2005-07-07', '7', 'true', '20.4'),
```
Comment (danmainah) [inline data.sql]: - [x] @MNisarAli kindly check on the date of birth, its suppose to be in June and not the month of July.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,91 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > '10.5';
...
+
+-- Update transaction
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
```
Comment (danmainah) [inline queries.sql:37]: - [x] @MNisarAli great work on your query, kindly add a query to verify the changes were made before commit, also update the screenshots to showcase that.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,91 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > '10.5';
...
+COMMIT;
+
+SELECT * FROM animals;
+
+-- Start transaction
+BEGIN;
+
+-- Delete all transactions
+DELETE FROM animals;
+
```
Comment (danmainah) [inline queries.sql:48]: - [x]  @MNisarAli great work on your query, kindly add a query to verify the changes were made before rollback, also update the screenshots to showcase that.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @MNisarAli , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
- [x] @MNisarAli kindly ensure your MIT link is working, currently its broken.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (MNisarAli) [reply data.sql]: Thanks, @danmainah, for bringing my attention to update the mistakenly added date of birth. 👍

**Review by christianbiring1 — APPROVED** [review]
Hi @MNisarAli , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**